src/providers/ipa/ipa_deskprofile_rules_util.c

```diff
@@ -0,0 +1,796 @@
+    if (ret != EOK) {
+        DEBUG(SSSDBG_CRIT_FAILURE,
+              "ipa_deskprofile_get_normalized_rule_name() failed [%d]: %s\n",
+              ret, sss_strerror(ret));
+        goto done;
+    }
+
+    ret = ipa_deskprofile_get_filename_path(tmp_ctx,
+                                            priority,
+                                            IPA_DESKPROFILE_RULES_USER_DIR,
+                                            domain,
+                                            username,
+                                            rule_prio,
+                                            user_prio,
+                                            group_prio,
+                                            host_prio,
+                                            hostgroup_prio,
+                                            normalized_rule_name,
+                                            extension,
+                                            &filename_path);
+    if (ret != EOK) {
+        DEBUG(SSSDBG_CRIT_FAILURE,
+              "ipa_deskprofile_get_filename_path() failed [%d]: %s\n",
+              ret, sss_strerror(ret));
+        goto done;
+    }
+
+    fp = fopen(filename_path, "w");
+    if (fp == NULL) {
+        ret = errno;
+        DEBUG(SSSDBG_TRACE_FUNC,
+              "Failed to create the Desktop Profile rule file \"%s\" "
+              "[%d]: %s\n",
+              filename_path, ret, sss_strerror(ret));
+        goto done;
+    }
+
+    fprintf(fp, "%s", data);
+
+    ret = fclose(fp);
+    if (ret != EOK) {
+        ret = errno;
+        DEBUG(SSSDBG_TRACE_FUNC,
+              "Failed to close the Desktop Profile rule file \"%s\" [%d]: %s\n",
+              filename_path, ret, sss_strerror(ret));
+        goto done;
+    }
+
+    ret = chown(filename_path, uid, gid);
+    if (ret != EOK) {
+        ret = errno;
+        DEBUG(SSSDBG_CRIT_FAILURE,
+              "Failed to own the Desktop Profile Rule file \"%s\" [%d]: %s\n",
+              filename_path, ret, sss_strerror(ret));
+        goto done;
+    }
+
+    ret = EOK;
+
+done:
+    talloc_free(tmp_ctx);
+    return ret;
+}
+
+errno_t
+ipa_deskprofile_rules_remove_user_dir(const char *user_dir)
+{
+    TALLOC_CTX *tmp_ctx;
+    DIR *dir;
+    errno_t ret;
+
+    tmp_ctx = talloc_new(NULL);
+    if (tmp_ctx == NULL) {
+        return ENOMEM;
+    }
+
+    dir = opendir(user_dir);
+    if (dir == NULL) {
+        ret = errno;
+        if (ret == ENOENT) {
+            ret = EOK;
+
+            goto done;
+        }
+
+        DEBUG(SSSDBG_TRACE_FUNC,
+              "Unable to open the directory \"%s\" [%d]: %s\n",
+              user_dir, ret, sss_strerror(ret));
+
+        goto done;
+    }
+
+    for (;;) {
+        struct dirent *ent;
+        struct stat info;
+        char *filepath;
+
+        ent = readdir(dir);
+        if (ent == NULL) {
+            break;
+        }
+
+        filepath = talloc_asprintf(tmp_ctx, "%s/%s", user_dir, ent->d_name);
```

Comment:
I've got rid of this function entirely as I could re-use/create stuff from files.c

So, fixed locally!